### PR TITLE
tests, container-disk: use safe expect-batcher

### PR DESCRIPTION
**Pull request by alonSadan**
**What this PR does / why we need it**:
https://github.com/kubevirt/kubevirt/pull/3882/files#diff-a5a178b4c66835110ceff63c61d28c96
introduced a safer expect-batcher, which reduces the chance of asserintg
"leftovers" from previous commands, and by that making the expect-batcher
less prone to bugs. Further details and use-cases can be found in the
PR and in the links mentioned in It's description.

This PR is a part of a series of PRs, for refactoring all
the test-suits to use the safe one.

Also `LoggedInAlpineExpecter` function form utils package
had been refactored.

Related PRs:
https://github.com/kubevirt/kubevirt/pull/3972
https://github.com/kubevirt/kubevirt/pull/3967

Signed-off-by: alonSadan <asadan@redhat.com>

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: tests/container_disk_test.go [hunk 1]
```diff
@@ -207,13 +207,16 @@ var _ = Describe("[rfe_id:588][crit:medium][vendor:cnv-qe@redhat.com][level:comp
 				Expect(err).ToNot(HaveOccurred(), "expected alpine to login properly")
```
Comment (RamLavi) [inline tests/container_disk_test.go:207]: I think there is a typo in the commit message `form`-->`from`.

Also, is it ok to add links to gihub PRs in our git repo commit message?
I mean, if tomorrow we decide to move this repo to gitlab (or the like) - these PR links will have no meaning.
in the PR description it's ok and good, but I think we should refrain from adding it the actual git commit description

File: tests/utils.go [hunk 2]
```diff
@@ -3004,19 +3004,18 @@ func LoggedInAlpineExpecter(vmi *v1.VirtualMachineInstance) (expect.Expecter, er
 	}
 
 	b := append([]expect.Batcher{
-		&expect.BSnd{S: "\n"},
 		&expect.BSnd{S: "\n"},
 		&expect.BExp{R: "localhost login:"},
 		&expect.BSnd{S: "root\n"},
-		&expect.BExp{R: "localhost:~\\#"}})
-	res, err := expecter.ExpectBatch(b, 180*time.Second)
+		&expect.BExp{R: "localhost:~#"}})
```
Comment (RamLavi) [inline tests/utils.go:3010]: can you explain why this change is needed?

File: tests/container_disk_test.go (hunk 1 above)
Comment (alonSadan) [reply tests/container_disk_test.go:207]: Fixed the typo.

Thanks for the explanation about the links in the commit message.
will remove those.

File: tests/utils.go (hunk 2 above)
Comment (alonSadan) [reply tests/utils.go]: `#` is not a special character and therefore does not needs escaping. so this double slash is redundant.
In fact, a new [commit](https://github.com/kubevirt/kubevirt/pull/3883/commits/fdbf06fc96e75e8dc3d3305706bcd518918c272d) was recently merged which generalizes the prompt  experssion.
so I'll change it to use that one instead of  `"localhost:~#"`

**Comment by alonSadan** [comment]
covered  here https://github.com/kubevirt/kubevirt/pull/4010
